spec/builders/online_application_builder_spec.rb

```diff
@@ -0,0 +1,43 @@
+# coding: utf-8
+require 'rails_helper'
+
+describe OnlineApplicationBuilder do
+
+  let(:current_time) { Time.zone.now }
+  let(:submission) { attributes_for :public_app_submission }
+
+  describe '#build' do
+    subject(:build_submission) { described_class.new(submission).build }
+
+    it 'returns non persisted OnlineApplication' do
+      is_expected.to be_a(OnlineApplication)
+      is_expected.not_to be_persisted
+    end
+
+    describe 'email contact' do
+      subject(:email) { build_submission.email_address }
+      describe 'when applicant responds `no`' do
+        it { is_expected.to eql nil }
+      end
+
+      describe 'when applicant responds `no`' do
+        let(:submission) { attributes_for :public_app_submission, :email_contact }
+
+        it { is_expected.to eql 'foo@bar.com' }
+      end
+    end
+
+    describe 'children field' do
```

Comment:
The same applies to here as above, the field will already be prepared on the public app side, so you'll only have a number in `children` field.